lib/SILOptimizer/IPO/ClosureSpecializer.cpp

```diff
@@ -225,13 +234,16 @@ class CallSiteDescriptor {
   SingleValueInstruction *
   createNewClosure(SILBuilder &B, SILValue V,
                    llvm::SmallVectorImpl<SILValue> &Args) const {
-    if (auto *PA = dyn_cast<PartialApplyInst>(getClosure()))
-      return B.createPartialApply(getClosure()->getLoc(), V, {}, Args,
+    if (auto *PA = dyn_cast<PartialApplyInst>(getClosure())) {
+      auto subst = PA->getSubstitutionMap().subst(
+          B.getFunction().getForwardingSubstitutionMap());
```

Comment:
I'm surprised this `subst()` call has any effect. The forwarding substitution map is the identity.

Reply:
I'm pretty sure it does. Because the `partial_apply` is not in the same function. So we need to substitute the old `partial_appy`'s substitution map with the new (cloned) function that we're building _this_ `partial_apply` in. Otherwise, we'll get types that don't exist in this (cloned) function.